src/Types/UnbalancedTransaction.purs

```diff
@@ -0,0 +1,69 @@
+module Types.UnbalancedTransaction where
+
+import Data.BigInt (BigInt)
+import Data.Map (Map)
+import Data.Maybe (Maybe)
+import Data.Newtype (class Newtype)
+import Prelude
+import Types.ByteArray (ByteArray(..))
+import Types.POSIXTimeRange (POSIXTimeRange)
+import Types.Transaction (Ed25519KeyHash(..), Transaction, TransactionHash, Value)
+
+newtype PubKey = PubKey ByteArray
+
+derive instance Newtype PubKey _
+derive newtype instance Eq PubKey
+
+newtype PaymentPubKey = PaymentPubKey PubKey
+
+derive instance Newtype PaymentPubKey _
+derive newtype instance Eq PaymentPubKey
+
+newtype TxOutRef = TxOutRef
+  { id :: TransactionHash
+  , index :: BigInt
+  -- ^ Index into the referenced transaction's outputs
+  }
+
+derive instance Newtype TxOutRef _
+derive newtype instance Eq TxOutRef
+
+newtype ValidatorHash = ValidatorHash ByteArray
+
+derive instance Newtype ValidatorHash _
+derive newtype instance Eq ValidatorHash
+
+newtype DatumHash = DatumHash ByteArray
```

Comment:
Similarly to what @vvtran commented on, we could just use `DataHash` from `Types.Transaction`